### ISSUE TITLE
[Performance] `FP16` model execution and convolution kernel time is slower then `FP32`

### ISSUE THREAD

**Issue by seungtaek94** [microsoft/onnxruntime#13838]
### Describe the issue

Hi. I try to inference resnet101 `FP16` model using onnxruntime.

When using large batch size(upper then 8), the `Model run` time is faster then `FP32` model, but the `Convolution kernel` time is slow. So if I use small batch size `model_run` time also slower then `FP32` model. Is it normal.?

I measure the second step inference time that after 1st step as warm-up using onnxruntime profiler and below table is the result.

|type|batchsize|model_run time|SequentialExecutor::Execut|Conv_106_kernel_time|
|------|---|---|---|---|
|FP32|16|43.645|16.664|0.151|
|FP16|16|24.058|19.16|0.131|
|FP32|1|14.875|14.304|0.160|
|FP16|1|19.761|19.383|0.178|

I upload model and profiling results file in [Google drive](https://drive.google.com/drive/folders/1aH4TmLMB9_36swar8s2HSJ62zAVu0Db4?usp=sharing).

It is tested on `windows10` with `rtx 2080 super` and `cuda` provider.

### To reproduce

I get and convert onnx model using below code.

```python
import torch
import onnxsim
import onnx
from torchvision.models import resnet101
from onnxconverter_common.float16 import convert_float_to_float16_model_path


if __name__ == "__main__":
    model = resnet101()
    model.eval()
    
    
    torch_input = torch.randn(16, 3, 224, 224)
    
    torch_output = model(torch_input)
    
    torch.onnx.export(model,
                      torch_input,
                      "resnet101_bn-dynamic.onnx",
                      input_names=["input"],
                      output_names=["output"],
                      dynamic_axes={"input":{0 :'batch'},
                                    "output":{0 :'batch'}})
    
    sim_model, check_ok = onnxsim.simplify("resnet101_bn-dynamic.onnx")
    
    onnx.save(sim_model, "resnet101_bn-dynamic_sim.onnx")
    
    fp16_model = convert_float_to_float16_model_path("resnet101_bn-dynamic_sim.onnx")
    onnx.save(fp16_model, 'resnet101_bn-dynamic_sim_fp16.onnx')  
```

### Test code
```python
import onnxruntime as ort
import numpy as np
import time
from tqdm.rich import tqdm


if __name__ == "__main__":
    sess_options = ort.SessionOptions()

    # Set graph optimization level
    sess_options.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_BASIC 
    sess_options.enable_profiling = True
    
    """
    model_fp32 = ort.InferenceSession("./resnet101_bn-dynamic_sim.onnx", sess_options, providers=["CUDAExecutionProvider"]) 
    tensor_fp32 = np.random.randn(16, 3, 224, 224).astype(np.float32)
    output = model_fp32.run(None, {"input": tensor_fp32}) # warmup    
    s = time.time()
    for i in tqdm(range(10)):
        output = model_fp32.run(None, {"input": tensor_fp32})
    e = time.time()
    print(f"Fp32: {e-s}sec")
    prof_file_fp32 = model_fp32.end_profiling()    
    
    
    """
    model_fp16 = ort.InferenceSession("./resnet101_bn-dynamic_sim_fp16.onnx", sess_options, providers=["CUDAExecutionProvider"])
    tensor_fp16 = np.random.randn(16, 3, 224, 224).astype(np.float16)
    output = model_fp16.run(None, {"input": tensor_fp16}) # warmup    
    s = time.time()
    for i in tqdm(range(10)):
        output = model_fp16.run(None, {"input": tensor_fp16})
    e = time.time()
    print(f"Fp16: {e-s}sec")    
    prof_file_fp16 = model_fp16.end_profiling()    
```

### Platform

Windows

### OS Version

Windows 10 Pro, 21H2, 19044.1826

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.11.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

CUDA

### Execution Provider Library Version

CUDA 11.4

### Model File

[Google drive](https://drive.google.com/drive/folders/1aH4TmLMB9_36swar8s2HSJ62zAVu0Db4?usp=sharing)

### Is this a quantized model?

Yes

**Comment by seungtaek94** [comment]
@tianleiwu  Thank you for your response :)
